src/api/app/models/kiwi/image.rb

```diff
@@ -37,9 +37,8 @@ class Image < ApplicationRecord
     validate :check_use_project_repositories
     validate :check_package_groups
     validates :preference, presence: true
-    accepts_nested_attributes_for :preference, reject_if: proc { |attributes|
-      attributes['type_containerconfig_name'].blank? && attributes['type_containerconfig_tag'].blank?
```

Comment:
who is that related to the version? 🤔

Reply:
Because before we only had containerconfiguration in the preference model and we rejected the nested attributes if it didn't contain a containerconfig name or tag. Now we also have a version in the preference so this is not valid anymore.

Reply:
because the preferences xml is like this:
```xml
  <preferences>
    <type image="docker" boot="grub">
      <containerconfig
        name="my_container"
        tag="latest" />
        <oemconfig>test</oemconfig>
    </type>
    <bootsplash-theme>gnome</bootsplash-theme>
    <bootloader-theme>gnome-dark</bootloader-theme>
    <version>2.0</version>
  </preferences>
```

And before we only stored the containerconfig name & tag so it didn't make sense to save preferences if there were no containerconfig attributes set. Now thats different since we also store preferences.version.